### DISCUSSION TITLE
Share options between commands by grouping options into a class

### DISCUSSION THREAD

**Discussion by aantn** [fastapi/typer#854] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class CommonOptions:
    name: str = typer.Option(..., help="Common name option")
    age: int = typer.Option(30, help="Common age option")

@app.command()
def command1(common: CommonOptions):
    typer.echo(f"Command 1: Name: {common.name}, Age: {common.age}")

@app.command()
def command2(common: CommonOptions):
    typer.echo(f"Command 2: Name: {common.name}, Age: {common.age}")

if __name__ == "__main__":
    app()
```

### Description

I know this has been discussed to death, but I'm not happy with any of the options. Would you be open to a PR that makes the attached code snippet supported?

### Operating System

macOS

### Typer Version

Not relevant

### Python Version

Not relevant

**Comment by svlandeg** [comment]
> I know this has been discussed to death, but I'm not happy with any of the options. 

Hi @aantn, can you provide the links to previous discussions, to have the full context? Thanks!

**Reply by aantn** [reply]
Sure.

* https://github.com/tiangolo/typer/issues/296
* https://github.com/tiangolo/typer/issues/153
* https://github.com/tiangolo/typer/issues/405

The solution that is closest to what I want is https://github.com/tiangolo/typer/issues/296#issuecomment-2121078281 but I'd like to improve on the syntax by adding native support.

**Comment by aantn** [comment]
I recently found https://github.com/greendwin/typer_di which seems to meet my requirements. I am still evaluating, but if typer_di works well, I will close this.

**Comment by aantn** [comment]
I ended up not using typer_di, but I think this would be an elegant solution: https://github.com/tiangolo/typer/discussions/665
